### ISSUE TITLE
Name, Title and other fields

### ISSUE THREAD

**Issue by pegaso1976ch** [yairEO/tagify#292]
Hi, first of all I have to confess that I'm really apreciating your code.

Now I'm trying to work on it to adapt it to my needs.

I'm tryingo to use the mix content and combining it with a database in which is contained a table with many peoples as shown below:
<table>
<tr>
<th>ID</th>
<th>Name</th>
<th>Company</th>
</tr>
<tr>
<td>1</td>
<td>Frank White</td>
<td>Microsoft</td>
</tr>
<tr>
<td>2</td>
<td>Michael Jackson</td>
<td>Sonz</td>
</tr><tr>
<td>3</td>
<td>James Brown</td>
<td>Apple</td>
</tr></table>

What I would like to do is to show the names on the list but return the ID.

<del>I would like also to change the tags interpolator form  [[  ]]  to <per></per>, I found the mixTagsInterpolator position but also if I change it, it don't work.</del> This part is ok, I reloaded the cache and it worked fine :)

**Comment by yairEO** [comment]
Sure, I will help you once I finish my day-job and get home. 

Hopefully in 7 hours.

**Comment by pegaso1976ch** [comment]
OK, good work :)

**Comment by lee-matthews** [comment]
Hi, like pegaso1976ch I require the same functionality. I'm equally working on a project that uses a database. I'd like to be able to show names in the textarea and recover each name's corresponding database id code when saving data from my textarea element.

Thanks for this great tool!

**Comment by yairEO** [comment]
Hey fine people of the internet, I understand your need and am contemplating on how to make this feasible, because there might be more than one person with the same name, and that should be represented in the textarea's pre-interpolated value somehow.

The [demo shows](https://yaireo.github.io/tagify/#section-mix):

   <textarea name='mix'>[[cartman]] and [[kyle]] do not know [[homer simpson]] because he's a relic.</textarea>

But this is not enough to actually know who's who if two people have the same name. 

I will need to make a smarter interpolation template.

Maybe a full Object, like so:

    <textarea>
      [[{"name":"Jack", "id":123}]] and [[{"name":"Jack", "id":987}]] are friends
    </textarea>

---

@pegaso1976ch - I need you to explain to me:

1. Are you sending to the server the whole *textarea* value?
2. You need to save, on the server, the `string` of the *textarea*, to be able to put it back in inside the *textarea* after page refresh.

So, as I've said, you must save the whole string on the server, but you **also** said you want to send to the server the tags data (only tags without regular text), so that can be achieved by accessing your instances `value` property: `tagify.value`<br>
It will return an Array of Objects (tags)

**Comment by pegaso1976ch** [comment]
@yairEO it's a little bit more complicated my procedure :) I use a contenteditable <div>, but this don't change the base of all.

Yes in my actual app I send the entire content of the textare to the processing page that stores the content of the field to a text file. The content is a thing like this:

This is the text on the <div> field
![1](https://user-images.githubusercontent.com/54842236/64233775-3f8c9d00-cef5-11e9-8c2d-c8c3bb1f67f2.jpg)

This is the text on the textfile:
![3](https://user-images.githubusercontent.com/54842236/64233831-5af7a800-cef5-11e9-91c4-d4f89ecee1e0.jpg)

On the end page the `<per>` code is replaced with the name of the person, with a javascript title that show all the information about the person, as shown here:
![2](https://user-images.githubusercontent.com/54842236/64233843-64811000-cef5-11e9-993d-a4e86d32a6a1.jpg)

At the moment I select the person from the select field over the <div> and the system appends the ID with the `<per>` tags to the end of the textfield it works,but this is obviously intricated.

**Comment by yairEO** [comment]
I will publish very soon a new version which supports this new format:

```html
<textarea>
  [[{"name":"Jack", "id":123}]] and [[{"name":"Jack", "id":987}]] are friends
</textarea>
```

---

Also, you could define your own custom Tag template, and there you are free to do your transformation from ID to a name. Here's how it might look like:

```html
<textarea>
  [[123]] and [[987]] are friends
</textarea>
```

```javascript
// I imagine you have a mapping technique of your own, and this is the simplest example:
var mapIdToName = {
	123 : "jack",
	987 : "veronica"
}

var tagify = new Tagify(..., {
    templates : {
        tag : function(v, tagData){
            try{
                return `<tag title='${v}' contenteditable='false' spellcheck="false" class='tagify__tag ${tagData.class ? tagData.class : ""}' ${this.getAttributes(tagData)}>
                    <x title='remove tag' class='tagify__tag__removeBtn'></x>
                    <div>
                        <span class='tagify__tag-text'>
                            ${mapIdToName[v] ? mapIdToName[v] : v}
                        </span>
                    </div>
                </tag>`
            }
            catch(err){}
        }
    }
}
```

**Comment by yairEO** [comment]
I've made a [demo page](https://jsbin.com/beruzil/edit?html,js,output) but then discovered a problem, when more than one tag with the same `value` exists in the textarea it will not be converted into a tag because Tagify's minor bug. I will fix it now and deploy a new version

**Comment by yairEO** [comment]
That's it, fixed on v `2.28.4`

**Comment by pegaso1976ch** [comment]
Thank you very much, I added some row on your <script> so that also when you are tiping the code works ;) 

`<script>

var whitelist_1 = [
  {value:"Jack", id:123},
  {value:"Kevi", id:987}
]

var input = document.querySelector('textarea');

var settings = {
  mode: 'mix',
  pattern: /@/,  // <--  Text starting with @ or # (if single, String can be used here)
  enforceWhitelist: true,
  whitelist: whitelist_1.map(function(item){ return typeof item == 'string' ? {value:item} : item}),
  duplicates: true,
  dropdown : {
            enabled: 1
        }

}
var tagify = new Tagify(input, settings);
</script>
`

Now, next question, how can I use this code with an editable `<div>`, something like this:

`<div  contenteditable="true" name="testo" id="testo"></div>`
